src/proteus/dswx_hls.py

```diff
@@ -1744,18 +1737,63 @@ def _compute_and_apply_cloud_layer(wtr_2_layer, fmask,
         logger.info(error_msg)
         raise Exception(error_msg)
 
-    logger.info(f'mask adjacent to cloud/cloud-shadow mode:'
-                f' {mask_adjacent_to_cloud_mode}')
- 
     # Check Fmask cloud shadow bit (3) => bit 0
-    cloud_layer[np.bitwise_and(fmask, 2**3) == 2**3] = 1
+    preliminary_cloud_layer[np.bitwise_and(fmask, 2**3) == 2**3] = 1
 
     if mask_adjacent_to_cloud_mode == 'mask':
         # Check Fmask adjacent to cloud/shadow bit (2) => bit 0
-       cloud_layer[np.bitwise_and(fmask, 2**2) == 2**2] = 1
+       preliminary_cloud_layer[np.bitwise_and(fmask, 2**2) == 2**2] = 1
 
     # Check Fmask cloud bit (1) => bit 2
-    cloud_layer[np.bitwise_and(fmask, 2**1) == 2**1] += 4
+    preliminary_cloud_layer[np.bitwise_and(fmask, 2**1) == 2**1] += 4
+
+    return preliminary_cloud_layer
+
+
+def _compute_and_apply_cloud_layer(wtr_2_layer, cloud_layer, fmask,
```

Comment:
Thank you for you suggestion. I'm trying to change the code as minimum as possible. I had to divide the original function into two parts because cloud/cloud shadow values were needed at the beginning of the execution (to populate the CLOUD_COVERAGE metadata of all DSWx-HLS layer) and the second part that needed the WTR-2 layer. I'd like to avoid adding another division to the original function.  To avoid confusion, I'm renaming `_compute_and_apply_cloud_layer()` to `_add_snow_to_cloud_layer_and_apply_cloud_masking()`.